astropy/io/misc/ecsv.py

```diff
@@ -209,57 +216,85 @@ class ECSVEngine:
         Dictionary mapping engine names to their respective engine classes.
     """
 
-    name = None
-    format = None
-    engines = {}
+    name: str | None = None
+    format: str | None = None
+    engines: dict[str, "ECSVEngine"] = {}
 
     def __init_subclass__(cls, **kwargs):
         """Register the subclass as an ECSV engine."""
         super().__init_subclass__(**kwargs)
+
+        # Ensure that the subclass has the required string class attributes.
+        for attr in ("name", "format"):
+            val = getattr(cls, attr, None)
+            if not isinstance(val, str):
+                raise TypeError(
+                    f"Subclasses of ECSVEngine must define a class attribute '{attr}' "
+                    f"as a string, got {type(val)}."
+                )
```

Comment:
If the error message is defined outside the class definition, then this can be reduced to

```suggestion
        if not isinstance(cls.name, str):
            raise TypeError(ERR_MSG.format("name", type(cls.name)))
        if not isinstance(cls.format, str):
            raise TypeError(ERR_MSG.format("format", type(format)))
```

Reply:
Alternatively... using a `_check_attr_is_a_str` helper method

```suggestion
        cls._check_attr_is_a_str("name")
        cls._check_attr_is_a_str("format")
```